pkg/idutil/imagewalker/imagewalker.go

```diff
@@ -68,17 +73,23 @@ func (w *ImageWalker) Walk(ctx context.Context, req string) (int, error) {
 	// to handle the `rmi -f` case where returned images are different but
 	// have the same short prefix.
 	uniqueImages := make(map[digest.Digest]bool)
-	for _, image := range images {
+	nameMatchIndex := -1
+	for i, image := range images {
 		uniqueImages[image.Target.Digest] = true
+		// to get target image index for `nerdctl rmi <short digest ids of another images>`.
+		if (parsedReferenceStr != "" && image.Name == parsedReferenceStr) || image.Name == req {
+			nameMatchIndex = i
```

Comment:
is it possible `name match` happens on multiple images?

Reply:
Thank you for your feedback! 

Yes, name matching can occur for multiple images if they share a similar prefix. 
For example, when there are several alpine images with different tags, running `nerdctl rmi alpine` will specifically remove `alpine:latest` by default:

```bash
> nerdctl images | grep alpine
alpine                              latest      beefdbd8a1da    About a minute ago    linux/arm64    10.46MB    4.09MB
alpine                              20190408    8b6b8c0f71e8    16 minutes ago        linux/arm64    6.844MB    2.708MB
alpine                              20190228    6199d795f07e    21 minutes ago        linux/arm64    6.844MB    2.706MB
ghcr.io/stargz-containers/alpine    3.13-org    ec14c7992a97    19 hours ago          linux/arm64    6.865MB    2.714MB

> nerdctl rmi alpine
Untagged: docker.io/library/alpine:latest@sha256:beefdbd8a1da6d2915566fde36db9db0b524eb737fc57cd1367effd16dc0d06d
Deleted: sha256:16113d51b7181f20135f51e8ffbaead20a7671cd783017601f198748ce8a8ebf

> nerdctl images | grep alpine
alpine                              20190408    8b6b8c0f71e8    22 minutes ago    linux/arm64    6.844MB    2.708MB
alpine                              20190228    6199d795f07e    27 minutes ago    linux/arm64    6.844MB    2.706MB
ghcr.io/stargz-containers/alpine    3.13-org    ec14c7992a97    19 hours ago      linux/arm64    6.865MB    2.714MB
```

However, if an image matches the given prefix, such as `alpine:custom`, the image must be removed by specifying `alpine:<tag name>`.

Does this example and explanation address your concern?